android/build.gradle

```diff
@@ -42,62 +46,54 @@ android {
     }
 
-    buildTypes {
-        buildTypes.each {
-            def autoMd = auto ? 'com.google.android.gms.car.application' : 'disabled.car.application'
-            it.manifestPlaceholders = [autoMetadata: autoMd]
-        }
+    compileOptions {
+        sourceCompatibility JavaVersion.VERSION_1_8
+        targetCompatibility JavaVersion.VERSION_1_8
     }
 }
 
 repositories {
+    google()
+    jcenter()
     maven {
         // As RN is not updated in jcenter anymore, we'll have to grab it from npm
         url "../node_modules/react-native/android"
     }
-    mavenCentral()
-    jcenter()
 }
 
 dependencies {
     //noinspection GradleDynamicVersion
-    compile 'com.facebook.react:react-native:+'
+    implementation('com.facebook.react:react-native:+') {
+        exclude group: 'com.android.support', module: 'support-compat'
+        exclude group: 'com.android.support', module: 'appcompat-v7'
+    }
 
     // ExoPlayer Core
-    if(exoplayer) {
-        compile 'com.google.android.exoplayer:exoplayer-core:r2.4.3'
-    } else {
-        provided 'com.google.android.exoplayer:exoplayer-core:r2.4.3'
-    }
+    implementation 'com.google.android.exoplayer:exoplayer-core:2.7.1'
 
     // ExoPlayer DASH
     if(dash) {
-        compile 'com.google.android.exoplayer:exoplayer-dash:r2.4.3'
+        implementation 'com.google.android.exoplayer:exoplayer-dash:2.7.1'
     } else {
-        provided 'com.google.android.exoplayer:exoplayer-dash:r2.4.3'
+        compileOnly 'com.google.android.exoplayer:exoplayer-dash:2.7.1'
     }
 
     // ExoPlayer HLS
     if(hls) {
-        compile 'com.google.android.exoplayer:exoplayer-hls:r2.4.3'
+        implementation 'com.google.android.exoplayer:exoplayer-hls:2.7.1'
     } else {
-        provided 'com.google.android.exoplayer:exoplayer-hls:r2.4.3'
+        compileOnly 'com.google.android.exoplayer:exoplayer-hls:2.7.1'
     }
 
     // ExoPlayer SmoothStreaming
     if(smoothstreaming) {
-        compile 'com.google.android.exoplayer:exoplayer-smoothstreaming:r2.4.3'
-    } else {
-        provided 'com.google.android.exoplayer:exoplayer-smoothstreaming:r2.4.3'
-    }
-
-    // Caching (for the default Android player)
-    if(cache && !exoplayer) {
-        compile 'com.danikula:videocache:2.7.0'
+        implementation 'com.google.android.exoplayer:exoplayer-smoothstreaming:2.7.1'
     } else {
-        provided 'com.danikula:videocache:2.7.0'
+        compileOnly 'com.google.android.exoplayer:exoplayer-smoothstreaming:2.7.1'
     }
 
-    // Make sure we're using at least the support library 23.2.0, not 23.0.1
-    compile 'com.android.support:appcompat-v7:23.2.0'
+    implementation "com.android.support:support-compat:$supportVersion"
+    implementation "com.android.support:appcompat-v7:$supportVersion"
+    implementation "com.android.support:support-media-compat:$supportVersion"
+    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
```

Comment:
I've been thinking a lot about using Kotlin in this project. Although it does have a few advantages, you still need to bundle a few dependencies just to use it. I don't think it's worth it in React Native, as you're already bundling a lot of native and JS code